### PR TITLE
Add PHPStan to the CI and fix some issues in tests

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Fix a bunch of problems in our test suite using static analysis.

### CODE REVIEW

File: tests/Bridge/Doctrine/Orm/Filter/SearchFilterTest.php [hunk 1]
```diff
@@ -9,7 +9,7 @@
  * file that was distributed with this source code.
  */
 
-namespace ApiPlatform\Core\Tests\Doctrine\Orm\Filter;
+namespace ApiPlatform\Core\Tests\Bridge\Doctrine\Orm\Filter;
```
Comment (theofidry) [inline tests/Bridge/Doctrine/Orm/Filter/SearchFilterTest.php]: I'm surprised the PSR-2 fixer didn't detect that...

**Comment by ondrejmirtes** [comment]
@dunglas Have you considered adding PHPStan to .travis.yml so that it's checking the source code continuously?

BTW: You can find [more about PHPStan here in this article](https://medium.com/@ondrejmirtes/phpstan-2939cd0ad0e3) and [on GitHub](https://github.com/phpstan/phpstan).

**Comment by dunglas** [comment]
@ondrejmirtes yes I'll definitely do it, but there are some remaining errors to handle before (mostly false positives or things we don't want to handle).

**Comment by ondrejmirtes** [comment]
@dunglas You can also solve that by running PHPStan on lower level (that's what they're for) or ignoring the messages with regexps in ignoreMessages as you do.

File: tests/Bridge/Doctrine/Orm/Filter/SearchFilterTest.php (hunk 1 above)
Comment (teohhanhui) [inline tests/Bridge/Doctrine/Orm/Filter/SearchFilterTest.php]: @theofidry Why would it?

File: tests/Metadata/Property/Factory/ExtractorPropertyMetadataFactoryTest.php [hunk 2]
```diff
@@ -65,7 +66,7 @@ public function testCreateWithNonexistentResourceXml()
     {
         $configPath = __DIR__.'/../../../Fixtures/FileConfigurations/resourcenotfound.xml';
 
-        (new ExtractorPropertyMetadataFactory(new XmlExtractor([$configPath])))->create(\ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\ThisDoesNotExist::class, 'foo');
+        (new ExtractorPropertyMetadataFactory(new XmlExtractor([$configPath])))->create(ThisDoesNotExist::class, 'foo');
```
Comment (teohhanhui) [inline tests/Metadata/Property/Factory/ExtractorPropertyMetadataFactoryTest.php]: Can't this be:

```php
'ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\ThisDoesNotExist'
```

instead?

Comment (dunglas) [inline tests/Metadata/Property/Factory/ExtractorPropertyMetadataFactoryTest.php]: It can indeed.

**Comment by ondrejmirtes** [comment]
What`s in PHPStan's dev-master that's not in 0.6.4 for compatibility? I'm quite against using dev-master because it can break your build quite easily.

**Comment by dunglas** [comment]
@ondrejmirtes https://github.com/phpstan/phpstan/pull/161

**Comment by ondrejmirtes** [comment]
@dunglas I see. But I think it's safer to require 0.6.4 (rather than having PHPStan break your build and make you angry) and ignore the scalar error for now. There will be quite a lot of changes for 0.7 and will likely to find new errors in applications related to nullables but it will still be work in progress for a while.

**Comment by dunglas** [comment]
@ondrejmirtes indeed, depending of a non-stable version is one of the reason preventing this PR to be merged for now.

File: tests/Mock/FosUser/MailerMock.php [hunk 3]
```diff
@@ -9,7 +9,7 @@
  * file that was distributed with this source code.
  */
 
-namespace ApiPlatform\Core\Tests\Mock;
+namespace ApiPlatform\Core\Tests\Mock\FosUser;
```
Comment (meyerbaptiste) [inline tests/Mock/FosUser/MailerMock.php]: You have to update this too: https://github.com/api-platform/core/blob/0bbf5c9154497983e6bff2a0e9bc81ef9a370f4e/tests/Fixtures/app/config/config.yml#L90

Comment (dunglas) [inline tests/Mock/FosUser/MailerMock.php]: done

File: src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php [hunk 4]
```diff
@@ -205,7 +205,7 @@ private function getSerializerGroups(string $resourceClass, array $options, stri
             $request = $this->requestStack->getCurrentRequest();
         }
 
-        if (null !== $request) {
+        if (null !== $this->serializerContextBuilder && null !== $request) {
```
Comment (soyuka) [inline src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php]: nice catch

File: .travis.yml [hunk 5]
```diff
@@ -36,6 +37,7 @@ script:
   - if [[ $coverage = 1 ]]; then phpdbg -qrr phpcov.phar merge --clover build/logs/clover.xml build/cov; fi
   - tests/Fixtures/app/console api:swagger:export > swagger.json && swagger validate swagger.json && rm swagger.json
   - if [[ $lint = 1 ]]; then php php-cs-fixer.phar fix --dry-run --diff --no-ansi; fi
+  - if [[ $lint = 1 ]]; then phpstan analyse -c phpstan.neon -l5 --ansi src tests; fi
```
Comment (teohhanhui) [inline .travis.yml]: Level 5! :tada:

**Pull request merged**